### ISSUE TITLE
in Tb128 - popup for restricted features: notification leads to exception + icon not displayed

### ISSUE THREAD

**Issue by RealRaven2000** [RealRaven2000/SmartTemplate4#308]
This problem is caused by the function `NotificationBox.appendNotification()` having been converted to async in mozilla-central 128 (Firefox' codebase)

```
notifyComposeBodyReady
 chrome://smarttemplate4/content/smartTemplate-util.js:640 
newNotification.messageImage is undefined
 popupLicenseNotification
  chrome://smarttemplate4/content/smartTemplate-util.js:640:7
insertTemplate@chrome://smarttemplate4/content/smartTemplate-compose.js:1659:16
smartTemplate-util.js:760:11
```

For example, when the user a variable like `%file()%`, without a valid SmartTemplates license, they get a notification such as the one shown below:

![image](https://github.com/user-attachments/assets/d4e37419-0fd7-4161-a4b6-12e84cdf10f5)

on top of that the button [Buy a License] overflows in the panel...

**Comment by RealRaven2000** [comment]
Here is a quick fix:

[smartTemplate-fx-4.6.2pre15.zip](https://github.com/user-attachments/files/16488497/smartTemplate-fx-4.6.2pre15.zip)

shows the correct notification and throws no more exceptions:

![image](https://github.com/user-attachments/assets/e712252d-fa38-45e1-9bf5-44d19ad4a2ad)


----

To try the version above download zip file and drag the file into Thunderbird Add-ons Manager (do not extract contents, it won't install like that)

**Comment by RealRaven2000** [comment]
Fixed in 4.7 - published 05/08/2024
